logic_mixin.py

```diff
@@ -238,33 +249,55 @@ def extract_submission_text_from_generated_text(self, prompt, generated_text):
 			# There must be at least a complete title to make a submission
 			return {}
 
-		if generated_text.startswith('<|sols|>'):
-			submission_type = 'url'
-			maintext_start_tag = '<|sol|>'
-			maintext_end_tag = '<|eol|>'
-		elif generated_text.startswith('<|soss|>'):
-			submission_type = 'selftext'
-			maintext_start_tag = '<|sost|>'
-			maintext_end_tag = '<|eost|>'
-
-		# Find the start and end tags of the main text, starting from the end of the title
-		idx_maintext_start = generated_text.find(maintext_start_tag, idx_title_end)
-		idx_maintext_end = generated_text.find(maintext_end_tag, idx_title_end)
-
-		if idx_maintext_start != (idx_title_end + len(title_end_tag)):
-			# check that the main text immediately follows the title end
-			return {}
-
-		if idx_maintext_start == -1 or idx_maintext_end == -1:
-			# There must be a complete body (link or submission)
-			return {}
-
 		title = generated_text[idx_title_start + len(title_start_tag):idx_title_end]
 
 		if not (0 < len(title) < 300):
 			# Validate the title length is within reddit's range
 			return {}
 
-		maintext = generated_text[idx_maintext_start + len(maintext_start_tag):idx_maintext_end]
+		# The title is ok, add it to the dict to return
+		return_dict['title'] = title
+
+		if generated_text.startswith('<|soss|>'):
+			selftext_start_tag = '<|sost|>'
+			selftext_end_tag = '<|eost|>'
+
+			# Find the start and end tags of the main text, starting from the end of the title
+			idx_selftext_start = generated_text.find(selftext_start_tag, idx_title_end)
+			idx_selftext_end = generated_text.find(selftext_end_tag, idx_title_end)
+
+			if idx_selftext_start == -1 or idx_selftext_end == -1:
+				# Both tags should be found
+				return {}
+
+			print(idx_selftext_start, (idx_title_end + len(title_end_tag)))
+			if idx_selftext_start != (idx_title_end + len(title_end_tag)):
+				# check that the main text immediately follows the title end
+				return {}
+
+			return_dict['selftext'] = generated_text[idx_selftext_start + len(selftext_start_tag):idx_selftext_end]
+
+		return return_dict
+
+	def find_image_url_for_search_string(self, search_string):
+		logging.info(f"Searching on Bing for an image for "{search_string}"")
```

Comment:
Sure! Looks like these quotes need to be escaped:
```suggestion
		logging.info(f"Searching on Bing for an image for \"{search_string}\"")
```

Reply:
Addressed in https://github.com/zacc/ssi-bot/pull/29/commits/96a94eb25101235339332f361f7ed17814c0a350